apache/nuttx
# use hostfs read/write on stdio as uart

## Summary
This is used to implement serial console on instruction set simulator, which don't have serial controller.
## Impact

## Testing
The cadence simulator is propriotery closed software, here is how to verify the patch with sim:

1. apply the PR change
2. change `arch/sim/src/sim/sim_uart.c` to remove `host_uart_start` , `uart_register` and `up_putc`
3. change `boards/sim/sim/sim/configs/nsh/defconfig`, add `CONFIG_FS_HOSTFS=y` `CONFIG_UART_HOSTFS=y`
4. build and run
```
 ./tools/configure.sh -E sim:nsh
 make -j8
 ./nuttx 
```
